src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -93,22 +139,38 @@ internal static SafeFreeCredentials AcquireCredentialsHandle(string package, boo
 
         internal static void ValidateImpersonationLevel(TokenImpersonationLevel impersonationLevel)
         {
-            throw new PlatformNotSupportedException();
+            if (impersonationLevel != TokenImpersonationLevel.Identification)
+            {
+                throw new ArgumentOutOfRangeException("impersonationLevel", impersonationLevel.ToString(),
+                    SR.net_auth_supported_impl_levels);
+            }
+
         }
 
         internal static void ThrowCredentialException(long error)
         {
-            throw new PlatformNotSupportedException();
+            string message = SR.net_auth_alert;
+            if (error == LogonDenied)
+            {
+                message = SR.net_auth_bad_client_creds;
+            }
+
+            if ((int)error == NegoState.ERROR_TRUST_FAILURE)
+            {
+                message = SR.net_auth_context_expectation_remote;
+            }
+
+            throw new AuthenticationException(message, null);
         }
 
         internal static bool IsLogonDeniedException(Exception exception)
         {
-            throw new PlatformNotSupportedException();
+            return exception.HResult == LogonDenied;
         }
 
         internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
         {
-            throw new PlatformNotSupportedException();
+            return new Win32Exception(NTE_FAIL);
```

Comment:
Yes the protocol requires a WIn32 error code for any server (even if Linux) implementing MS-NNS